tests/integration/common/block_utils.rs

```diff
@@ -167,13 +167,23 @@ pub fn os_hints(
     transactions: Vec<InternalTransaction>,
     tx_execution_infos: Vec<TransactionExecutionInfo>,
 ) -> (StarknetOsInput, ExecutionHelperWrapper) {
-    let mut contracts: HashMap<Felt252, ContractState> = blockifier_state
-        .state
-        .address_to_class_hash
-        .keys()
+    let state_diff = blockifier_state.to_state_diff();
+    let deployed_addresses = state_diff.address_to_class_hash.keys().cloned().collect::<HashSet<_>>();
+
+    let initial_addresses = blockifier_state.state.address_to_class_hash.keys().cloned().collect::<HashSet<_>>();
+
+    let addresses = initial_addresses.union(&deployed_addresses);
+
+    let mut contracts: HashMap<Felt252, ContractState> = addresses
+        .into_iter()
         .map(|address| {
+            let contract_hash = if deployed_addresses.contains(address) {
+                Felt252::ZERO
```

Comment:
Why do you set it by default? Is it because it doesn't exist? Shouldn't this be an `Option` type instead?

Reply:
It is exepected by the cairo code: https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/starknet/core/os/execution/deprecated_execute_syscalls.cairo#L742

Reply:
Added a comment.